packages/cli/src/lib/cmds/build.ts

```diff
@@ -30,137 +30,170 @@ interface SigningKeyPasswords {
+    try {
+      const manifestFile = path.join(process.cwd(), 'twa-manifest.json');
+      const twaManifest = await TwaManifest.fromFile(manifestFile);
+      const pwaValidationResult =
+          await PwaValidator.validate(new URL(twaManifest.startUrl, twaManifest.webManifestUrl));
+      return Result.ok(pwaValidationResult);
+    } catch (e) {
+      return Result.error(e);
     }
+  }
+
+  async generateAssetLinks(
+      twaManifest: TwaManifest, passwords: SigningKeyPasswords): Promise<void> {
+    try {
+      const digitalAssetLinksFile = './assetlinks.json';
+      const keyInfo = await this.keyTool.keyInfo({
+        path: twaManifest.signingKey.path,
+        alias: twaManifest.signingKey.alias,
+        keypassword: passwords.keyPassword,
+        password: passwords.keystorePassword,
+      });
+
+      const sha256Fingerprint = keyInfo.fingerprints.get('SHA256');
+      if (!sha256Fingerprint) {
+        this.log.warn('Could not find SHA256 fingerprint. Skipping generating "assetlinks.json"');
+        return;
+      }
 
-    const digitalAssetLinks =
-      DigitalAssetLinks.generateAssetLinks(twaManifest.packageId, sha256Fingerprint);
+      const digitalAssetLinks =
+        DigitalAssetLinks.generateAssetLinks(twaManifest.packageId, sha256Fingerprint);
 
-    await fs.promises.writeFile(digitalAssetLinksFile, digitalAssetLinks);
+      await fs.promises.writeFile(digitalAssetLinksFile, digitalAssetLinks);
 
-    log.info(`Digital Asset Links file generated at ${digitalAssetLinksFile}`);
-    log.info('Read more about setting up Digital Asset Links at https://developers.google.com' +
-        '/web/android/trusted-web-activity/quick-start#creating-your-asset-link-file');
-  } catch (e) {
-    log.warn('Error generating "assetlinks.json"', e);
+      this.log.info(`Digital Asset Links file generated at ${digitalAssetLinksFile}`);
+      this.log.info('Read more about setting up Digital Asset Links at https://developers.google.com' +
+          '/web/android/trusted-web-activity/quick-start#creating-your-asset-link-file');
+    } catch (e) {
+      this.log.warn('Error generating "assetlinks.json"', e);
+    }
   }
-}
 
-export async function build(config: Config, args: ParsedArgs,
-    log = new Log('build'), prompt: Prompt = new InquirerPrompt): Promise<boolean> {
-  let pwaValidationPromise;
-  if (!args.skipPwaValidation) {
-    pwaValidationPromise = startValidation();
+  async buildApk(signingKey: SigningKeyInfo, passwords: SigningKeyPasswords): Promise<void> {
+    await this.gradleWrapper.assembleRelease();
+    await this.androidSdkTools.zipalign(
+        './app/build/outputs/apk/release/app-release-unsigned.apk', // input file
+        './app-release-unsigned-aligned.apk', // output file
+    );
+    const outputFile = './app-release-signed.apk';
+    await this.androidSdkTools.apksigner(
+        signingKey.path,
+        passwords.keystorePassword, // keystore password
```

Comment:
The these comments needed? Your variables are named well enough to be self explanatory.

Reply:
done.